### ISSUE TITLE
Touch-delay on large archive

### ISSUE THREAD

**Issue by perry-mitchell** [buttercup/buttercup-mobile#66]
My personal archive is quite large and when initially unlocking, it's impossible to scroll the list of groups for several seconds. Taps activate the item underneath after the delay.

EDIT: Potentially consider [FlatList](https://facebook.github.io/react-native/docs/flatlist.html) as a replacement for performance reasons. This will require an RN upgrade.

**Comment by farfromrefug** [comment]
I tested your app on android and i have to say it's really slow and my archive is not even that big (a few hundred items).
Obviously this is due the react-native implementation of the ListView. They don't want to use the full power of the native recyclerview/uicollectionview and instead decided to use full JS implementation.
And this won't be fixed in a near future...
Have you considered other alternative to react-native? Beeing a mobile dev i have been using my own fork of Titanium mobile for a few years. But even if i can make a full blown app in no time (and you won't be able to say it's not native!) i think the future is with NativeScript.
Have you ever looked at it?
If you were open to consider it i would like to help. I am thinking about writing a  NativeScript version of Buttercup.

**Comment by perry-mitchell** [comment]
Hi @farfromrefug,

Yes I agree that the existing listview is slow - I too only have a couple of hundred items and it lags for some reason. I am hopeful that there are ways around this performance issue without the need to change to another platform. The [FlatList](https://facebook.github.io/react-native/docs/flatlist.html) component also looks quite inviting here from a performance perspective. Let's see.

We did consider a few alternatives before starting: NativeScript, React-Native, Titanium, Phone Gap etc. (I've even built a couple of apps in Titanium, but didn't enjoy it). I've not used NativeScript and can't really put any words on it.. but I would suggest that we stick with the current RN build for some very strong reasons:

 1. The app is already working - not much point to rewrite it just yet, if there is even a need
 2. We're using react+redux which is quite commonplace across our stack
 3. Several internal packages that we can use cross platform (React based)

There's been an enormous amount of work put into the current app - and not because of React-Native - but because doing JS-based crypto for Buttercup is hard on mobile. There's a lot of performance complexities which we had to overcome. I don't think it's wise to tackle those again without great reason.

I'm always happy to consider better alternatives, but personally I don't love how NativeScript lays out their apps (from only looking at their examples - happy to be shown better ones). I prefer the React way of working, for sure.

I hope this explains at least my position on this - I'd love to see other attempts at Buttercup applications, but adoption of new builds may be difficult. I'd love to first try to fix these issues in RN and if it goes to hell some time down the road then we'll obviously look for a contender to replace the mobile app we have.

**Comment by farfromrefug** [comment]
Thanks for the explanation.
I understand the react approach. We actually had the same approach with one of my client where we have an electron app just like you.
As a side note about Titanium, I didn't like it when I started with it a few years ago but it was the only one at the time why use actual native code while levering the power of js. That s why I décided to make it my own. Now I really start to see its limitation especially around v8 and android. That s why I started looking at other solutions.
First I tried react native and pushed it a bit to its limits to see what it could do. I even digged deep into the code. Dont expect much of the nativelist it is still not native and still tries to do it on the js side.
Now about Nativescript I can understand the fact that you miss the react layout. And I really understand the ease for your devs to code all apps using the same system.
But honestly the power of Nativescript is quite amazing. It can use angular but indeed no react...
If I manage to find time I might try at porting your app to nativescript it should be quite straightforward. It could give you a chance to test it
A final note about crypto. There something great you could do which is use the native crypto from node on desktop/web and make a simple native shim o that mobile(so easy with nativescript...). This would boost perfs for sure.

I have a lot of interest in buttercup because I am looking for a 1password alternative for me and my company. I need a solution for all my colleagues. I like your desktop app but the mobile app is kind of a show stopper for now. But being open source there are a lot of solutions and ways to help :)
Thanks again for your answer

**Comment by perry-mitchell** [comment]
Closing as V2 is out, and the details in this issue are not relevant to the current build. 

If something like this occurs again, please open a new issue and reference this one.
